src/Component/Customer/CustomerSelector.php

```diff
@@ -29,26 +30,33 @@ class CustomerSelector implements CustomerSelectorInterface
     protected $session;
 
     /**
-     * @var \Symfony\Component\Security\Core\SecurityContextInterface
+     * @var \Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface
      */
-    protected $securityContext;
+    protected $tokenStorage;
+    
+    /**
+     * @var \Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface
+     */
+    protected $authorizationChecker;
 
     /**
      * @var string
      */
     protected $locale;
 
     /**
-     * @param CustomerManagerInterface $customerManager
-     * @param SessionInterface         $session
-     * @param SecurityContextInterface $securityContext
-     * @param LocaleDetectorInterface  $localeDetector
+     * @param CustomerManagerInterface      $customerManager
+     * @param SessionInterface              $session
+     * @param TokenStorageInterface         $tokenStorage
+     * @param AuthorizationCheckerInterface $authorizationChecker
+     * @param LocaleDetectorInterface       $localeDetector
      */
-    public function __construct(CustomerManagerInterface $customerManager, SessionInterface $session, SecurityContextInterface $securityContext, LocaleDetectorInterface $localeDetector)
+    public function __construct(CustomerManagerInterface $customerManager, SessionInterface $session, TokenStorageInterface $tokenStorage, AuthorizationCheckerInterface $authorizationChecker, LocaleDetectorInterface $localeDetector)
```

Comment:
yes, I'm editing it according to what jordisala1991 said in https://github.com/sonata-project/ecommerce/pull/441#issuecomment-299504486 (I will push after some hours)

I have to add a parameter $authorizationChecker (without type hinting) as was done in https://github.com/sonata-project/SonataUserBundle/blob/6f37cdf9dc57c0c71ba3b01a1856dfd6f9c686fc/Security/EditableRolesBuilder.php  or you have another solution?